src/graph/visitor/PrunePropertiesVisitor.cpp

```diff
@@ -57,7 +57,7 @@ void PrunePropertiesVisitor::visitCurrent(Project *node) {
     rootNode_ = false;
     return;
   }
-  for (auto i = 0u; i < columns.size(); ++i) {
```

Comment:
Add `DCHECK_EQ(colNames.size(), columns.size())`

Reply:
this check may not be true in the current implementation. there are cases that their sizes differ. this current pr would fix the crash problem first.

Reply:
Should fix the size of colName or columns, or the result is incorrect.

Reply:
ok, let me check it then. any clue why they were not the same?

Reply:
```
  // last vertex
  DCHECK(!nodeInfos.empty());
  addNode(nodeInfos.back());
```

`MatchSolver::buildProjectColumns` didn't set col name for the last vertex in the path.

Reply:
With more investigation, the error is expected to come from `CollapseProjectRule::transform`.

Reply:
Update the cause of the problem to `PushFilterDownProjectRule::transform`.

Reply:
A `filter->project-filter` group was optimized into `filter->project` after pushing the last filter completely down to the first one. The new project's output variable was set to the old and disappeared filter's output variable, which have different column names, cauding the columns and column names in the new project to differ.

This is the direct cause of the size differ that crashes graphd while prunning the properties, which does not check for size differences before this PR.

Reply:
The filter was improperly setup by the `PushFilterDownAggregateRule`. This shall be the final root cause.